esrally/client/factory.py

```diff
@@ -336,8 +336,9 @@ def cluster_distribution_version(hosts, client_options, client_factory=EsClientF
     """
     # no way for us to know whether we're talking to a serverless elasticsearch or not, so we default to the sync client
     es = client_factory(hosts, client_options).create()
-    # unconditionally wait for the REST layer - if it's not up by then, we'll intentionally raise the original error
-    wait_for_rest_layer(es)
+
+    # wait_for_rest_layer  calls the Cluster Health API, which is not available for unprivileged users on Serverless
```

Comment:
Can we please update the docstring to include the new `serverless_operator` return value?